plugin/src/main/java/org/opensearch/ml/rest/RestMLUpdateModelAction.java

```diff
@@ -54,17 +55,21 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
      */
     private MLUpdateModelRequest getRequest(RestRequest request) throws IOException {
         if (!request.hasContent()) {
-            throw new IOException("Model update request has empty body");
+            throw new OpenSearchParseException("Model update request has empty body");
         }
 
         String modelId = getParameterId(request, PARAMETER_MODEL_ID);
 
         XContentParser parser = request.contentParser();
         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-        MLUpdateModelInput input = MLUpdateModelInput.parse(parser);
-        // Model ID can only be set here. Model version can only be set automatically.
-        input.setModelId(modelId);
-        input.setVersion(null);
-        return new MLUpdateModelRequest(input);
+        try {
+            MLUpdateModelInput input = MLUpdateModelInput.parse(parser);
+            // Model ID can only be set here. Model version can only be set automatically.
+            input.setModelId(modelId);
+            input.setVersion(null);
+            return new MLUpdateModelRequest(input);
+        } catch (IllegalStateException e) {
+            throw new OpenSearchParseException(e.getMessage());
+        }
```

Comment:
Is there an UT test missing for this catch statement? You can provide a null input in the request and check the return Exceptions.

Reply:
There is a UT test.